be/src/exprs/vectorized/array_functions.tpp

```diff
@@ -230,4 +229,103 @@ private:
         }
     }
 };
+
+template <PrimitiveType PT>
+class ArrayReverse {
+public:
+    using ColumnType = RunTimeColumnType<PT>;
+    using CppType = RunTimeCppType<PT>;
+
+    static ColumnPtr process(FunctionContext* ctx, const Columns& columns) {
+        DCHECK_EQ(columns.size(), 1);
+
+        size_t chunk_size = columns[0]->size();
+
+        if (columns[0]->only_null()) {
+            return ColumnHelper::create_const_null_column(chunk_size);
+        }
+
+        ColumnPtr src_column = ColumnHelper::unpack_and_duplicate_const_column(chunk_size, columns[0]);
+        ColumnPtr dest_column = src_column->clone();
+
+        if (dest_column->is_nullable()) {
+            _reverse_array_column(down_cast<NullableColumn*>(dest_column.get())->mutable_data_column(), chunk_size);
+        } else {
+            _reverse_array_column(dest_column.get(), chunk_size);
+        }
+        return dest_column;
+    }
+
+private:
+    static void _reverse_fixed_column(Column* column, size_t begin, size_t end) {
+        auto& data = down_cast<ColumnType*>(column)->get_data();
+        std::reverse(data.begin() + begin, data.begin() + end);
+    }
+
+    static void _reverse_binary_column(Column* column, size_t begin, size_t end) {
+        auto& offsets = down_cast<BinaryColumn*>(column)->get_offset();
+        // revert size
+        std::reverse(offsets.begin() + begin + 1, offsets.begin() + end + 1);
+        // convert size to offset
+        for (size_t i = begin; i < end; i++) {
+            offsets[i + 1] = offsets[i] + offsets[i + 1];
+        }
+
+        // revert all byte of one array
+        auto& bytes = down_cast<BinaryColumn*>(column)->get_bytes();
+        std::reverse(bytes.begin() + offsets[begin], bytes.begin() + offsets[end]);
+
+        // revert string one by one
+        for (size_t i = begin; i < end; i++) {
+            std::reverse(bytes.begin() + offsets[i], bytes.begin() + offsets[i + 1]);
+        }
+    }
+
+    static void _reverse_data_column(Column* column, const Buffer<uint32_t>& offsets, size_t chunk_size) {
+        if constexpr (pt_is_fixedlength<PT>) {
+            for (size_t i = 0; i < chunk_size; i++) {
+                _reverse_fixed_column(column, offsets[i], offsets[i + 1]);
+            }
+        } else if constexpr (pt_is_binary<PT>) {
+            auto& data = down_cast<BinaryColumn*>(column)->get_offset();
+            // convert offset ot size
+            for (size_t i = data.size() - 1; i > 0; i--) {
```

Comment:
Would better put this for loop to `_reverse_binary_column` method, otherwise, the logic in `_reverse_binary_column ` for offsets column is confusing.

Reply:
Fixed